lib/chef/resource/chef_client_scheduled_task.rb

```diff
@@ -129,13 +139,22 @@ class ChefClientScheduledTask < Chef::Resource
 
         # According to https://docs.microsoft.com/en-us/windows/desktop/taskschd/schtasks,
         # the :once, :onstart, :onlogon, and :onidle schedules don't accept schedule modifiers
+
+        frequency_mod = if new_resource.frequency == "minute" && new_resource.frequency_modifier == 30
+                          30
+                        elsif new_resource.frequency_modifier != 30
+                          new_resource.frequency_modifier
+                        else
+                          1
+                        end
```

Comment:
This should probably be added as the default on the property, then tests could be written for all the different combos of inputs.

Reply:
just so I understand what you mean, you're saying the default value should be 1, and then have tests for the frequency being "minute" that would modify this to 30? rather than the other way around?  I think "minute" is the only frequency option that would be bad to end up with a default frequency modifier of 1.

Reply:
This logic should be used to determine the default and then we should have 

```ruby

Reply:
Sorry to be so vague. I meant this logic could be moved into the default of the `frequency_modifier` property. Here's an example of a property with a dynamic default that depends on the value of another parameter: https://github.com/chef/chef/blob/master/lib/chef/resource/execute.rb#L630-L632

With that changed, it'd be easier to test that the right values were being set depending on the input values.

Also, with a dynamic default a `default_description` should be added that describes the logic in English so that it'll show up in the autogenerated docs.

Reply:
oh slick! I was hoping something like that existed for the default.  I'll try to get an update to this today to make the logic better.

Reply:
New, better code:

```ruby
      property :frequency_modifier, [Integer, String],
        coerce: proc { |x| Integer(x) },
        callbacks: { "should be a positive number" => proc { |v| v > 0 } },
        description: "Numeric value to go with the scheduled task frequency",
        default: lazy { frequency == "minute" ? 30 : 1 }, 
        default_description: "30 if frequency is 'minute', 1 otherwise"
```

I believe this resolves this.  Thanks for the help @phiggins and @tas50